### ISSUE TITLE
Map with collection as input, instead of arguments

### ISSUE THREAD

**Issue by swistak35** [Tarrasch/zsh-functional#16]
What is really missing in shell which is very functional is treating input as an array.
It's different approach than current functions in this library.
For example, I have a file like this:

```
~% cat testlines 
foo
bar baz
jajco
```

And if I had some time ago following use case:

```
~% cat testlines | while read line; do echo $line | base64; done
Zm9vCg==
YmFyIGJhego=
amFqY28K
```

It would be great to run this like that:

```
~% cat testlines | mapl 'base64'
```

Do you think it would fine to include this in your library?

**Comment by NightMachinery** [comment]
I played around, and this seems to work:
```
each 'echo $1' "${(f@)$(<~/some-file)}"
```

**Comment by Tarrasch** [comment]
Yes. But we want a functional left-to-right feeling in this library. I promise I'm still gonna review any good pull requests. :)
